include/circt/Dialect/RTL/RTLCombinatorial.td

```diff
@@ -251,6 +251,48 @@ def ConcatOp : VariadicRTLOp<"concat"> {
   ];
 }
 
+def ArraySliceOp : RTLOp<"array_slice", [NoSideEffect]> {
+  let summary = "Get a range of values from an array";
+  let description = [{
+    Extracts a sub-range from an array. The range is from `lowIndex` to
+    `lowIndex` + the number of elements in the return type, non-inclusive on
+    the high end. For instance,
+
+    ```
+    // Slices 16 bytes starting at '%offset'
```

Comment:
A super trivial nit pick:  Please change this comment to "16 elements" since this is an element-based operation, it just happens to be bytes in this case.

Overall, this patch is super awesome, thanks again!